analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/ProgramState.cs

```diff
@@ -25,14 +25,27 @@ namespace SonarAnalyzer.SymbolicExecution.Roslyn
     public sealed record ProgramState : IEquatable<ProgramState>
     {
         public static readonly ProgramState Empty = new();
-        private ProgramState[] toArray;
+        private readonly ImmutableDictionary<IOperation, SymbolicValue> operationValue;
+        private readonly ImmutableDictionary<ISymbol, SymbolicValue> symbolValue;
+        private readonly ImmutableDictionary<CaptureId, IOperation> captureOperation;
+        private readonly ImmutableHashSet<ISymbol> preservedSymbols;
+        private readonly ImmutableStack<ExceptionState> exceptions;
 
-        private ImmutableDictionary<IOperation, SymbolicValue> OperationValue { get; init; }     // Current SymbolicValue result of a given operation
-        private ImmutableDictionary<ISymbol, SymbolicValue> SymbolValue { get; init; }
+        private ProgramState[] toArray;
+        private int? operationValueHashCode;
+        private int? symbolValueHashCode;
+        private int? captureOperationHashCode;
+        private int? preservedSymbolsHashCode;
+        private int? exceptionsHashCode;
+        private int? hashCode;
+
+        // Current SymbolicValue result of a given operation
```

Comment:
You can remove this. I don't find it useful anymore when the engine has evolved and the ProgramState is well understood.
```suggestion
```